Assets/Scripts/Layers/LayerTypes/GeoJSONLineLayer.cs

```diff
@@ -39,16 +43,57 @@ protected override void OnLayerActiveInHierarchyChanged(bool activeInHierarchy)
             LineRenderer3D.gameObject.SetActive(activeInHierarchy);
         }
 
-        public void AddAndVisualizeFeature(Feature feature, MultiLineString featureGeometry, CoordinateSystem originalCoordinateSystem)
+        public void AddAndVisualizeFeature<T>(Feature feature, CoordinateSystem originalCoordinateSystem)
+            where T : GeoJSONObject
         {
-            LineFeatures.Add(feature);
-            GeoJSONGeometryVisualizerUtility.VisualizeMultiLineString(featureGeometry, originalCoordinateSystem, LineRenderer3D);
+            // Skip if feature already exists (comparison is done using hashcode based on geometry)
+            if (SpawnedVisualisations.Any(f => f.feature.GetHashCode() == feature.GetHashCode()))
+                return;
+
+            var newFeatureVisualisation = new FeatureLineVisualisations() { feature = feature };
+
+            if (feature.Geometry is MultiLineString multiLineString)
```

Comment:
Please use methods for the activities in an if-else or switch statement for readability